graph-manager-docs/source/federation.mdx

```diff
@@ -123,9 +123,115 @@ server.listen().then(({ url }) => {
+#### When to call `service:push`
+
+In general, when calling `service:push`, our goal should always be to roll out changes in a way that is backwards-compatible to existing clients, in
+order to avoid any downtime. Because configuration changes can affect the query planner, it's highly recommended to **call `apollo service:push` after
+all replicas of an federated service have deployed**. This way, you can be certain that only valid operations will be executable against the graph, and
+that operations will not be able to access fields or types that do not yet exist.
+
+In rare cases where the schema changes cause a backwards incompatible change to _query-planning_, however, it's recommended to update the service registry
+**before** deploying federated services. In particular, schema changes that affect query planning should be done as separate changes to the registry, prior
+to other modifications. This will help avoid the situation that the query planner generates queries that fail validation in downstream services or violate
+resolvers' expectations. Some examples of this include:
+
+1. Modifying `@key` or `@requires` dependencies while simultaneously modifying dependency fields
+2. Removing a type implementation from an interface
+
+For example, with an interface that has implementations in multiple services, such as the following:
+
+```
+// Channel service
+interface Channel @key(fields: "id") {
+  id: ID!
+}
+
+// Web service
+type WebChannel implements Channel @key(fields: "id") {
+  id: ID!
+}
+
+// Email service
+type EmailChannel implements Channel @key(fields: "id") {
+  id: ID!
+}
+```
+
+In order to remove the `EmailChannel` implementation of the `Channel` interface, you'll want to perform the following steps:
+
+1. Register a new `email` service that removes the `EmailChannel` implementation of `Channel`
+1. Deploy a new Email service that removes the implementation
+
+The first step will cause the query-planner to stop sending along fragments `...on EmailChannel`, which would fail validation if sent
+to a service that did not have awareness of that type!
+
+#### Modifying query-planning logic
+
+You'll want to treat migrations of query planning logic similarly to how you may treat database migrations, by carefully considering the effects on the
+downstream services as the query planner changes, and planning for "double reading" as appropriate. For example, consider the case of migrating a
+`@required` field in an extension from one to another. In this example, we'll consider the `Products` service and the `Reviews` service, like so:
+
+```js
+// Products Service
+
+type Product @key(fields: "upc") {
+  upc: ID!
+  nameLowerCase: String!
+}
+
+// Review Service
+
+extend type Product @key(fields: "upc") {
+  upc: ID! @external
+  reviews: [Review]! @requires(fields: "nameLowercase")
+  nameLowercase: String! @external
+}
+```
+
+If the `nameLowercase` field is to be deprecated and replaced with the `name` field
```

Comment:
"If we want to deprecate the `nameLowercase` field and replace it with the `name` field:" (avoid passive)